datafusion/core/tests/user_defined/user_defined_scalar_functions.rs

```diff
@@ -434,15 +434,10 @@ impl ScalarUDFImpl for RandomUDF {
     }
 
     fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
-        let len: usize = match &args[0] {
-            // This udf is always invoked with zero argument so its argument
-            // is a null array indicating the batch size.
-            ColumnarValue::Array(array) if array.data_type().is_null() => array.len(),
-            _ => {
-                return Err(datafusion::error::DataFusionError::Internal(
-                    "Invalid argument type".to_string(),
-                ))
-            }
+        let len = if args.is_empty() {
+            1
+        } else {
+            return internal_err!("Invalid argument type");
```

Comment:
It might be good to have the test match the rest of the functions for consistency

```suggestion
        return exec_err!("Expect function to take no parameters");
```